### DISCUSSION TITLE
changing request parameters value in validator data()

### DISCUSSION THREAD

**Discussion by JaconsMorais** [adonisjs/core#2252] [Help]
Hello guys, I'm using data() to get id from params so I can validate him with one rule, here the code:

```
get data() {
    const requestBody = this.ctx.request.all()
    const uid = this.ctx.params.id

    return { ...requestBody, uid }
  }
```

and, in rules I created one regex rule to test if uid is one uuid valid string.

**this work**

But, I need to change one value in request, and I tried to use the same approach but did not work.

```
get data() {
    const requestBody = this.ctx.request.all()
    const data = {}
    data['uid'] = this.ctx.params.id

    if('rg' in requestBody) data['rg'] = '${requestBody['rg']}'.replace(/^\D+/g, '')

    return { ...requestBody, ...data }
  }
```

my intention is, if 'rg' is in request parameters, remove all chars that are not a number

But, this did not work, validate receives the uid, but receives the older value o 'rg'.

**Comment by artem-itdim** [comment]
You can use a new variable 

```javascript
get data() {
    const data = { ... this.ctx.request.all(), uid: this.ctx.params.id }

    if (data.rg) {
      data.rg = data.rg.replace(/^\D+/g, '')
    }

    return data
  }
```

**Reply by JaconsMorais** [reply]
Hi, thanks for your reply.
But this is not the exactly same thing I'm doing in my code?
